crates/bevy_solari/src/realtime/node.rs

```diff
@@ -0,0 +1,192 @@
+    },
+    renderer::{RenderContext, RenderDevice},
+    view::{ViewTarget, ViewUniform, ViewUniformOffset, ViewUniforms},
+};
+
+pub mod graph {
+    use bevy_render::render_graph::RenderLabel;
+
+    #[derive(Debug, Hash, PartialEq, Eq, Clone, RenderLabel)]
+    pub struct SolariLightingNode;
+}
+
+pub struct SolariLightingNode {
+    bind_group_layout: BindGroupLayout,
+    initial_and_temporal_pipeline: CachedComputePipelineId,
+    spatial_and_shade_pipeline: CachedComputePipelineId,
+}
+
+impl ViewNode for SolariLightingNode {
+    type ViewQuery = (
+        &'static SolariLighting,
+        &'static SolariLightingResources,
+        &'static ExtractedCamera,
+        &'static ViewTarget,
+        &'static ViewPrepassTextures,
+        &'static ViewUniformOffset,
+    );
+
+    fn run(
+        &self,
+        _graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        (
+            solari_lighting,
+            solari_lighting_resources,
+            camera,
+            view_target,
+            view_prepass_textures,
+            view_uniform_offset,
+        ): QueryItem<Self::ViewQuery>,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let pipeline_cache = world.resource::<PipelineCache>();
+        let scene_bindings = world.resource::<RaytracingSceneBindings>();
+        let view_uniforms = world.resource::<ViewUniforms>();
+        let frame_count = world.resource::<FrameCount>();
+        let (
+            Some(initial_and_temporal_pipeline),
+            Some(spatial_and_shade_pipeline),
+            Some(scene_bindings),
+            Some(viewport),
+            Some(gbuffer),
+            Some(depth_buffer),
+            Some(motion_vectors),
+            Some(view_uniforms),
+        ) = (
+            pipeline_cache.get_compute_pipeline(self.initial_and_temporal_pipeline),
+            pipeline_cache.get_compute_pipeline(self.spatial_and_shade_pipeline),
+            &scene_bindings.bind_group,
+            camera.physical_viewport_size,
+            view_prepass_textures.deferred_view(),
+            view_prepass_textures.depth_view(),
+            view_prepass_textures.motion_vectors_view(),
+            view_uniforms.uniforms.binding(),
+        )
+        else {
+            return Ok(());
+        };
+
+        let bind_group = render_context.render_device().create_bind_group(
+            "solari_lighting_bind_group",
+            &self.bind_group_layout,
+            &BindGroupEntries::sequential((
+                view_target.get_unsampled_color_attachment().view,
+                solari_lighting_resources.reservoirs_a.as_entire_binding(),
+                solari_lighting_resources.reservoirs_b.as_entire_binding(),
+                gbuffer,
+                depth_buffer,
+                motion_vectors,
+                view_uniforms,
+            )),
+        );
+
+        let frame_index = frame_count.0.wrapping_mul(5782582);
```

Comment:
What is the meaning of this number? Does it need to be _this_ number or a number with some specific properties? I think it deserves a comment

Reply:
Random number I chose, I think? I can't remember if I took it from a specific RNG technique or not. Maybe I did.

Reply:
Its very unlikely that this is a specific RNG technique's number because it is an even number, and does not occupy all the bits available evenly either. Looks like keyboardmash to me.